docs/AdvancedContributorGuidelines.md

```diff
@@ -0,0 +1,49 @@
+# Advanced Contributor Guidelines
+
+## Choosing issues
+
+* Follow issue priorities. Prefer to take higher priority issues over lower priority issues.
+* Favor issues related to certain features/aspects/techs to deepen expertise in those areas. 
+ 
+## Creating commits in big PRs
```

Comment:
@damithc 
> It's hard to define 'big' even in LoC because sometimes a trivial change (an invasive refactoring) can touch a lot of lines.

The thing is that effective reviewing of PRs either require that we either:

1. Eyeball each line of the diff line by line. (Skipping lines means potential bugs, or even security issues if the author tries to sneak a backdoor in. Yes, that happens.), or

2. Perform the same steps as the author (usually involves automated tools e.g. eclipse refactoring) and then compare (diff) our results.

So yes, I think LOC does matter. Too big a LOC count means that we can't use method (1) for reviewing, and if it can't be reviewed using method (2), then the PR author must absolutely break down the PR into smaller steps with smaller LOC.

> Perhaps it's the reviewer, realizing the PR is too big, who should ask the dev to refactor the commits to make it a multi-step PR?

Yes, that usually happens, but the point of putting this down in the documentation is so that contributors are aware of it. e.g. if contributors want to dump a gigantic PR with large LOC count without proper commit organization, they better have an explanation ready for why it is that way.